internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go

```diff
@@ -92,6 +93,37 @@ func (s *Server) Reboot(ctx context.Context, in *empty.Empty) (reply *machine.Re
 	return reply, nil
 }
 
+// Rollback implements the machine.MachineServer interface.
+//
+// nolint: dupl
+func (s *Server) Rollback(ctx context.Context, in *machine.RollbackRequest) (reply *machine.RollbackResponse, err error) {
+	log.Printf("rollback via API received")
+
+	if err := syslinux.Revert(); err != nil {
+		return nil, fmt.Errorf("failed to revert bootloader: %v", err)
+	}
+
+	go func() {
+		if err := s.Controller.Run(runtime.SequenceReboot, in, runtime.WithForce()); err != nil {
+			log.Println("reboot failed:", err)
+
+			if err != runtime.ErrLocked {
+				// NB: We stop the gRPC server since a failed sequence triggers a
+				// reboot.
+				s.server.GracefulStop()
+			}
+		}
+	}()
+
+	reply = &machine.RollbackResponse{
+		Messages: []*machine.Rollback{
+			{},
```

Comment:
Maybe the response should say which version we rolled back to?

Reply:
Couldn't hurt.